Feature valera

ValeriyKryatov opened the pull request:
1) Частично реализована обработка кнопок ТГ бота;
2) Реализованы сущности;
3) Подключена БД + написан скрипт создания таблиц БД

ValeriyKryatov committed:
Create README.md

ValeriyKryatov committed:
PetShelterTGBot.
1) Частично реализована обработка кнопок ТГ бота;
2) Реализованы сущности;
3) Подключена БД + написан скрипт создания таблиц БД

ValeriyKryatov committed:
Merge remote-tracking branch 'origin/feature-Valera' into feature-Valera

ValeriyKryatov committed:
PetShelterTGBot.
1) Занимаюсь переносом текстовой информации в отдельные константы

ValeriyKryatov committed:
PetShelterTGBot.
1) Занимаюсь переносом текстовой информации в отдельные константы
2) Реализую функционал кнопок для приюта собак

ValeriyKryatov committed:
PetShelterTGBot.
1) Занимаюсь переносом текстовой информации в отдельные константы
2) Реализую функционал кнопок для приюта собак
Продолжение

ValeriyKryatov committed:
PetShelterTGBot.
1) Занимаюсь переносом текстовой информации в отдельные константы
2) Реализую функционал кнопок для приюта собак
Внесены еще изменения в функционал

alekseykoshelev commented on src/main/java/PetShelterTGBot/service/TelegramBot.java:
```diff
@@ -0,0 +1,243 @@
+                messageText = "/come back" + animalsFlag.getTitle();
+            }
+            actionSelectorFromUpdate(messageText, chatId);
+            System.out.println("  Определили имя пользователя бота в update.getMessage().hasText() ==> " + nameUser);
```
Тут можно использовать логгер, вывод будет немного удобнее

alekseykoshelev commented on src/main/java/PetShelterTGBot/service/TelegramBot.java:
```diff
@@ -0,0 +1,243 @@
+            sendPhotoRequest.setPhoto(inputFile);
+//            sendPhotoRequest.setPhoto(inputFile2);
+            this.execute(sendPhotoRequest);
+        } catch (NullPointerException | TelegramApiException e) {
```
лучше сделать проверку через if в тех местах, где может быть null. Обычно NPE не ловят через try catch

alekseykoshelev commented on src/main/java/PetShelterTGBot/service/TelegramBot.java:
```diff
@@ -0,0 +1,243 @@
+//            sendPhotoRequest.setPhoto(inputFile2);
+            this.execute(sendPhotoRequest);
+        } catch (NullPointerException | TelegramApiException e) {
+            e.printStackTrace();
```
Можно также вывести ошибку в логгер

ValeriyKryatov committed:
PetShelterTGBot.
Работаю над созданием REST запросов

ValeriyKryatov committed:
PetShelterTGBot.
1) Изменена зависимость в pom.xml для корректной работы со Swagger,
2) Реализованы контроллеры, сервисы, репозитории, исключения,
3) Внесены все актуальные изменения от Харитонова Александра,
4) Удалена кнопка /menu3 из меню в ТГ боте,

ValeriyKryatov committed:
PetShelterTGBot.
1) Реализованы контроллеры, сервисы, репозитории для сообщений волонтерам и о посещении приюта,
2) Внесены все актуальные изменения от Харитонова Александра,
3) Удалены неиспользуемые импорты,
4) Бот и REST-запросы в Swagger работают

ValeriyKryatov committed:
PetShelterTGBot. На проверку, бот готов и работает
